lib/loadImage.js

```diff
@@ -84,13 +98,16 @@ function getPngInfo(data, info, options) {
     var decode = options.decode || checkTransparency;
 
     if (decode) {
-        var decodedResults = PNG.sync.read(data);
-        info.decoded = decodedResults.data;
-        info.width = decodedResults.width;
-        info.height = decodedResults.height;
-        if (checkTransparency) {
-            info.transparent = hasTransparency(info);
-        }
+        return parsePng(data)
+            .then(function(decodedResults) {
+                info.decoded = decodedResults.data;
+                info.width = decodedResults.width;
+                info.height = decodedResults.height;
+                if (checkTransparency) {
+                    info.transparent = hasTransparency(info);
+                }
+                return info;
+            });
     }
     return info;
```

Comment:
I refactored this area a bit so this is no longer a problem.

However, this was okay because nothing was calling `.then` on the result, and a promise chain had already started before.